kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/EnvelopeEncryptionFilterIT.java

```diff
@@ -218,8 +221,12 @@ void produceAndConsumeEncryptedAndPlainTopicsAtSameTime(KafkaCluster cluster, To
         }
     }
 
+    // TODO express this test as a unit test and consider doing away with the test as the IT level.
     @TestTemplate
     void shouldGenerateOneDek(KafkaCluster cluster, Topic topic, TestKmsFacade testKmsFacade) throws Exception {
+        assumeThatCode(testKmsFacade::getKms).doesNotThrowAnyException();
```

Comment:
I'd really prefer that getKms wasn't part of the API at all.  The ITs shouldn't need to care.  I'd like to revisit this, but not block getting this merged and 0.4.0 out.